### DISCUSSION TITLE
On-demand ISR - how to leverage a custom solution from leerob/on-demand-isr example?

### DISCUSSION THREAD

**Discussion by mi-hol** [vercel/next.js#38385] [Help]
@leerob to continue our conversation from https://github.com/leerob/on-demand-isr/issues/1457 as you suggested.

on-demand-isr seems to be the feature I was looking for as a solution to a knowledge base editing process problem.
As a NOOB with NEXT.js and web-based development I have some questions regarding setup that are currently not answered by [README.md](https://github.com/leerob/on-demand-isr#readme)

What I'd like to accomplish is to create a simple editing & publish process for knowledge base content stored as .md files on github.

The current readme is a bit short for noobs because it seems that pre-requisites are not detailed enough so NOOBs could successfully create a working solution out of it.
I'd envisage a step-by-step guide detailing "how-to leverage a custom solution from the demo" using a kind of "site template" as the final result and assume that 80% of the required steps are already described in the current readme. 

With my current (very limited) understanding of NEXT.js what I assume to be missing are the following steps:
- prerequisites to build a custom solution (maybe just a pointer to an already existing doc)

1. setup custom web server with next.js
2. download code from https://github.com/leerob/on-demand-isr to custom web server directory
3. modify downloaded code using a specified list of files (i.e. /components/Layout.tsx, ... ) or a parameter file (final goal would be to enhanced current code over time using a template approach)
4. trigger build

my questions:
- does my proposal make sense?
- would it be feasible to collaborate joinly on creating this missing piece of doc?
- are my assumptions about "what is currently missing" heading in the right direction?
- any other comments?

**Comment by mi-hol** [comment]
Digging a bit deep into next.js I found https://swr.vercel.app/docs/getting-started 
Looks to me like this is the type of doc that needs to be created for On-demand ISR. 
@leerob Would you share this view?

**Comment by mi-hol** [comment]
I cloned on-demand-ISR demo and now get error: "1E08010C:DECODER routines::unsupported"

Searches point to an issue in OpenSSL resolved by using `export NODE_OPTIONS=--openssl-legacy-provider` but that does work.
Any suggestion on how to fix it?

```

> next dev

ready - started server on 0.0.0.0:3000, url: http://localhost:3000
info  - Loaded env from /home/user/TestApp/on-demand-isr/.env.local
event - compiled client and server successfully in 2s (169 modules)
wait  - compiling / (client and server)...
event - compiled client and server successfully in 909 ms (268 modules)
wait  - compiling /_error (client and server)...
event - compiled client and server successfully in 186 ms (269 modules)
error - Error: error:1E08010C:DECODER routines::unsupported
    at Sign.sign (node:internal/crypto/sig:131:29)
    at Object.sign (/home/user/TestApp/on-demand-isr/node_modules/jwa/index.js:152:45)
    at Object.jwsSign [as sign] (/home/user/TestApp/on-demand-isr/node_modules/jws/lib/sign-stream.js:32:24)
    at module.exports [as sign] (/home/user/TestApp/on-demand-isr/node_modules/jsonwebtoken/sign.js:204:16)
    at getGitHubJWT (webpack-internal:///./lib/github.ts:18:64)
    at setAccessToken (webpack-internal:///./lib/github.ts:59:17)
    at readAccessToken (webpack-internal:///./lib/github.ts:54:15)
    at getStaticProps (webpack-internal:///./pages/index.tsx:31:91)
    at Object.renderToHTML (/home/user/TestApp/on-demand-isr/node_modules/next/dist/server/render.js:451:26)
    at async doRender (/home/user/TestApp/on-demand-isr/node_modules/next/dist/server/base-server.js:912:38) {
  library: 'DECODER routines',
  reason: 'unsupported',
  code: 'ERR_OSSL_UNSUPPORTED',
  page: '/'
}

```

**Reply by leerob** [reply]
I'm guessing you might need to set up the correct GitHub tokens - did you follow the steps in the README?

**Reply by mi-hol** [reply]
Well I tried to follow the README but for any NOOB certain steps require knowledge to be done correctly that a NOOB simply wont have.
Therefore errors are to be expected and only meaningful error messages help to get over these hurdles.

The README doesn't reveal the version of major components used, hence I installed "latest" versions 
Looking at the identical error reported on  https://github.com/octokit/auth-app.js/issues/332 a version issue seems to be the root cause of my error too.
" downgraded to node16 and moved the Pkey to a file and it started working."

Could you please add versions of major components used successfully to README?

```
npm ll
on-demand-isr@
│ /home/user/TestApp/on-demand-isr
│
├── @types/jsonwebtoken@8.5.8
│
├── @types/node@18.0.3
│
├── @types/react@18.0.15
│
├── eslint-config-next@12.2.2-canary.0
│
├── eslint@8.18.0
│
├── highlight.js@11.5.1
│
├── javascript-time-ago@2.5.6
│
├── jsonwebtoken@8.5.1
│
├── marked@4.0.17
│
├── next@12.2.2-canary.0
│
├── postcss-hover-media-feature@1.0.2
│
├── react-dom@18.2.0
│
├── react@18.2.0
│
├── sass@1.53.0
│
└── typescript@4.7.4


```
